ANTLR/src/Compiler/ContextualAnalysis/TypeCheckerVisitor.java

```diff
@@ -199,6 +194,36 @@ public Integer visitFuncdef(FuncdefContext ctx) {
         return returnType;
     }
 
+    private <T extends ParserRuleContext> ArrayList<Integer> getListOfTypes(T node, Integer size){
+        switch (node.getRuleContext().getRuleIndex()) {
+            case BuffParser.RULE_exprparams:
+                return listCreatedByLoop((ExprparamsNotEmptyContext)node, size);
+            case BuffParser.RULE_funcdef:
+                return listCreatedByLoop((FuncdefContext)node, size);
+        };
+        return null;
+    }
```

Comment:
Jeg ville foretrække et direkte kald til listCreatedByLoop() i stedet for den her switch, **hvis det altså er muligt.**

Reply:
Vi skal kende til rulecontext. så alternativet er en chainet if statement, medminder du kender en ANTLR feature der kan hjælpe?